src/compiler/binder.ts

```diff
@@ -583,6 +583,21 @@ namespace ts {
             }
         }
 
+        function jsdocTreatAsExported(node: Node) {
+            if (!isJSDocTypeAlias(node)) return false;
+            // jsdoc typedef handling is a bit of a doozy, but to summarize, treat the typedef as exported if:
+            // 1. It has an explicit name (since by default typedefs are always directly exported, either at the top level or in a container), or
+            if (!isJSDocEnumTag(node) && !!node.fullName) return true;

```

Comment:
could this be written positively as

```suggestion
            if (isJSDocTypedefTag(node) && !!node.fullName) return true;
```
I think enum and typedef are the only two possibilities here. Oh wait, maybe callback is too.